core/tests/org.fusesource.ide.camel.model.service.core.tests.integration/src/main/java/org/fusesource/ide/camel/model/service/core/tests/integration/core/model/DataFormatsInCamelModelElementIT.java

```diff
@@ -78,4 +79,12 @@ public void testDataFormatsInUnmarshal_notRemovedWhenUpdatingRefParameter() thro
 		assertThat(((AbstractCamelModelElement) unmarshallElement.getParameter("dataFormatType")).getNodeTypeId()).isEqualTo("json");
 	}
 
+	@Test
+	public void testCanFindTypes() throws CoreException, IOException {
```

Comment:
The problem here is that this tests finding data formats from the main CamelModel class, not the DataFormatModel. I have attempted to instantiate the DataFormatModel in a test, but it doesn't get populated correctly.

Reply:
Also, this test will fail for many of the model versions. Not quite sure how to filter out which ones will work and which ones won't.